addons/website_sale/models/product_image.py

```diff
@@ -41,3 +41,9 @@ def _check_valid_video_url(self):
         for image in self:
             if image.video_url and not image.embed_code:
                 raise ValidationError(_("Provided video URL for '%s' is not valid. Please enter a valid video URL.") % image.name)
+
+    def create(self, vals):
+        new_context = dict(self.env.context).copy()
+        if 'product_variant_id' in vals[0] and 'default_product_tmpl_id' in self.env.context:
+            del new_context['default_product_tmpl_id']
```

Comment:
Instead of deleting it from the new context, I would only create the new context (without this key) if it has to be changed. This also avoids calling `with_context` when not necessary (even if it is less costly with the new ORM than before).

Something like `self = self.with_context({k: v for k, v in self.env.context.items() if k != 'default_product_tmpl_id'}` in the condition.